tools/osv-scanner/cmd/osv-scanner/config.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+type LoadConfigErrorType int64
+
+const (
+	GlobalConfigSet LoadConfigErrorType = iota
+	NoConfigFound
+)
+
+type Config struct {
+	IgnoredVulnIds []string
+}
+
+type LoadConfigError struct {
+	targetPath string
+	errorType  LoadConfigErrorType
+}
+
+func (e LoadConfigError) Error() string {
+	switch e.errorType {
+	case GlobalConfigSet:
+		return "Global config has been set"
+	case NoConfigFound:
+		return "No config file found on this or any ancestor path: " + e.targetPath
+	}
+	panic("Invalid error type")
+}
+
+// TryLoadConfig tries to load config in `target` or any of it's parent dirs
+// `target` will be the key for the entry in configMap
+// Will shortcut and return "" if globalConfig is not nil
+func TryLoadConfig(target string, configMap map[string]Config) (string, error) {
```

Comment:
Can we hide the details of this `configMap` completely? And make this a more modular struct with methods? As is, users (e.g. the `filterResponse` function in the other file) have to do complicated logic to use and understand this structure, when ideally they should just call a single function. 

i.e. something like: 

```golang 

type Config struct {
 ...
}

type ConfigManager struct {
  configMap map[...]...
   ...
}

func (c *Config) ShouldIgnore(vuln) bool {
}

func (c *ConfigManager) UseOverride(configPath string) {
   ...
}

func (c *ConfigManager) Get(targetPath string) Config {
   ... // caches things in configMap if needed. 
}
```